docs/en/stack/security/get-started-tls.asciidoc

```diff
@@ -0,0 +1,429 @@
+the file and key. If you plan to add more nodes to your cluster in the future, 
+retain a copy of the file and remember its password.
+
+Alternatively, you can use a commercial or organization-specific CA, though that 
+scenario is not covered in this tutorial. See the `csr` option in the {ref}/certutil.html[elasticsearch-certutil] command. 
+--
+
+. Generate a certificate and private key for each node in your cluster. 
++
+--
+If you had a multi-node cluster and you chose to do hostname verification, you 
+would need to create a separate certificate for each node. You can generate 
+certificates and keys for multiple nodes at once by specifying the `--multiple` 
+parameter, which prompts you for details about each node or instance.
+
+In this tutorial, however, we are only using a single node cluster, 
+// TBD: Is this step even need in a single-node cluster?
+so run the following command once from your {es} directory:
+
+["source","sh",subs="attributes,callouts"]
+----------------------------------------------------------------------
+./bin/elasticsearch-certutil cert --ca elastic-stack-ca.p12 \
+--dns <domain_name> <1> --ip <ip_addresses> <2> --name <file_name> <3>
+----------------------------------------------------------------------
+// NOTCONSOLE
+<1> A comma-separated list of DNS names for your {es} node.
+<2> A comma-separated list of IP addresses for your {es} node.
+<3> A unique name for the certificate.
+
+If you chose not to do hostname verification, you can omit the `--dns`, `--ip`, 
+and `--name` parameters. In that case, the certificate does not have hostname 
+information in it and you can use the same certificate for every node in your 
+cluster.
+
+This command prompts you for a password. You can enter a password for your 
+certificate and key, or you can leave the password blank by pressing *Enter*.
+
+The output from this command is a single PKCS#12 keystore that includes a node 
+certificate, node key, and CA certificate. You can alternatively generate 
+Privacy Enhanced Mail (PEM) formatted certificates and keys, though that option 
+is not covered in this tutorial. 
+--
+
+. Copy the node certificate to the {es} configuration directory. For example, 
+copy the `.p12` file that you created in the previous step into a `certs` folder 
+in the {es} `config` directory. If you had a multi-node cluster, you would 
+create the `certs` folder on each node and copy the appropriate node certificate 
+to each node. 
+
+. Create a certificate for use by {kib}. 
++
+--
+{kib} doesn't currently support PKCS#12 keystores, so we must create a PEM 
+formatted certificate for it. For example:
+
+["source","sh",subs="attributes,callouts"]
+----------------------------------------------------------------------
+openssl pkcs12 -in elastic-certificates.p12 -cacerts -nokeys -out elastic-ca.pem
+----------------------------------------------------------------------
+
+In this example, `elastic-certificates.p12` is the PKCS#12 keystore that you 
```

Comment:
In this example, `elastic-stack-ca.p12` is the PKCS#12 keystore containing the Elasticsearch cluster's CA.